Lib/uuid.py

```diff
@@ -668,20 +668,21 @@ def getnode():
     if _node is not None:
         return _node
 
-    if sys.platform == 'win32':
-        getters = [_windll_getnode, _netbios_getnode, _ipconfig_getnode]
-    else:
-        getters = [_unix_getnode, _ifconfig_getnode, _ip_getnode,
-                   _arp_getnode, _lanscan_getnode, _netstat_getnode]
+    if getters is None:
+        if sys.platform == 'win32':
+            getters = [_windll_getnode, _netbios_getnode, _ipconfig_getnode]
+        else:
+            getters = [_unix_getnode, _ifconfig_getnode, _ip_getnode,
+                       _arp_getnode, _lanscan_getnode, _netstat_getnode]
 
     for getter in getters + [_random_getnode]:
         try:
             _node = getter()
         except:
             continue
-        if _node is not None:
+        if (_node is not None) and (0 <= _node < (1 << 48)):
             return _node
-    assert False, '_random_getnode() returned None'
+    assert False, '_random_getnode() returned an invalid value'
```

Comment:
What about including the invalid value in the assertion message?  (probably with a `.format()` call)

This case can't possibly happen but if for some odd reason it does, it would be good to know what the value was.  Your test to the conditional means it could be something other than None.